src/generic/RelSeries.jl

```diff
@@ -234,6 +234,10 @@ function renormalize!(z::AbstractAlgebra.RelSeriesElem)
    return nothing
 end
 
+function characteristic(a::RelSeriesRing{T}) where T <: RingElement
+   return characteristic(base_ring(a))
+end
```

Comment:
Instead of having to define it for all series types unders the sun, it could be nice to define a generic `characteristic(a::AbstractAlgebra.SeriesRing) = characteristic(base_ring(a))` and be done for the definition for most (all?) of them. Migth apply to other abstract types as well.

Reply:
E.g `PolyRing`.

Reply:
I don't know if that would be correct. So I haven't done this.

It's not so obvious that it would be correct for all possible implementations. For example, the characteristic of GF(p) is p, but if you think of p-adics as being a power series with elements in GF(p) (with carries), which some package might actually do, this would be incorrect.

Probably I am worrying unnecessarily here. But I did not define it for all series types on purpose. I cannot imagine what we will consider to be a series type in future, let alone what others will. And characteristic is pretty important to get right.

Reply:
This makes some sense, as `characteristic` is currently not in the mandatory interface for `SeriesRing`. So someone implementing a `SeriesRing` subtype is certain to not have a wrong answer, it will just get a `MethodError` when `characteristic` is needed. The drawback is that it forces this someone to implement `characteristic` if he needs the functionality (e.g. for `^` in this case). So I find the alternative to define generically `characteristic(a::AbstractAlgebra.SeriesRing) = characteristic(base_ring(a))` more pragmatic, in the case we expect it to almost never be wrong:
1. It allows almost everybody to not have to implement it
2. On the other hand, it has to be put in the docs, something along the lines of: "if you implement a subtype of `SeriesRing`, you must implement `characteristic` if your type behaves weirdly".

But the fact that `^` requires `characteristic` to be implemented suggests that it should be put in the docs anyway. So I would favor allowing people to have to do less work, at the risk of getting the wrong answer if someone implements a weird type and doesn't implement the required interface for weird types.

Just my opinion though, I won't argue for it.

Reply:
There's no need to define characteristic for ^ to work. I structured the PR so that it won't give incorrect answers. You get the result faster if you can use pow_fps. But you can only use that if you have char 0. There are many rings for which you can't decide the characteristic.

The suggestion to have:

characteristic(S) = characteristic(coefficient_ring(S))

for both polynomial rings and series has been discussed with @fieker, and I believe we agree we can do this. That's much more mathematically meaningful and it will resolve the current inefficiency in some cases. But it's a separate PR. I'll now open a ticket for it. (Naturally there is already a lot to do to roll the current thing out across all our packages.)

Reply:
> There's no need to define characteristic for ^ to work

Ah sure, sorry I got confused about that.

Otherwise yes, this plan sounds good. I will review once more now to help have this merged fast.